### DISCUSSION TITLE
Issues building monorepo with type only imports due to top level await in backend code

### DISCUSSION THREAD

**Discussion by KilianB** [vercel/next.js#69251] [Help]
### Summary

When attempting to build a next application in a monorepo environment the next code imports types from the backend package.

````tsx
import type { LocationHist } from "../../../backend/src/indexNextBackend";
````

Building fails because next is evaluating all imports to indexNextBackend.ts itself and errors out on es2017 fetures like top level await.

````
../backend/src/db/cacheDb.ts:9:22
Type error: Top-level 'await' expressions are only allowed when the 'module' option is set to 'es2022', 'esnext', 'system', 'node16', 'nodenext', or 'preserve', and the 'target' 
option is set to 'es2017' or higher.

   7 | export const WEEK = DAY * 7;
   8 |
>  9 | export const redis = await createClient()
````

As I just learned here https://github.com/vercel/next.js/discussions/14390 changing the target in the tsconfig does not affect the next build output as it uses it's own compiler.

I understand that types get evaluated and followed in order to do proper linting, but I do not want next to dictate how my backend project is set up or looks like. Types can be extracted into separate file to solve this issue, but should next really enfore syntax that was introduced **15 years ago** ? 

If NextJs is using tsc under the hood for type evaluation is it possible to bump the target for type only import checks? Is the syntax checking really necessary here, or am I missing a build flag I can use to cirumvent this problem?

**Comment by Bluzzi** [comment]
Hello, do you find a solution for this problem ? @KilianB

**Comment by KilianB** [comment]
My solution was to move the types into a 3rd "shared" project which only contains types. Since these imports no longer touch code related to awaits it circumvents the issue. In my case this was fine since I wanted protobuf type definitions to be separated out anyways.
The overall gist is still the same. Next forced me to change the way how the overall project was set up.
